src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,192 @@
+	const [ isEditing, setIsEditing ] = useState( ! attributes.restaurantIDs.length );
+
+	const prevIDs = usePrevious( attributes.restaurantIDs );
+
+	const getLocaleCode = () => {
+		if ( typeof coblocksBlockData.localeCode !== 'undefined' ) {
+			return coblocksBlockData.localeCode;
+		}
+		return '';
+	};
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	//TODO: 400/500 error handling and show loading spinner
+	const searchRestaurants = useCallback( debounce( ( token ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.catch( () => {
+				setNoResultsFound( true );
+			} )
+			.then( ( json ) => {
+				const results = [];
+				for ( const item in json.items ) {
+					const itemProps = json.items[ item ];
+					const name = `${ decodeURIComponent( itemProps.name ) },  ${ decodeURIComponent( itemProps.addressResponse.city ) }, ${ decodeURIComponent( itemProps.addressResponse.country ) } (${ itemProps.rid })`;
+					results.push( name );
+				}
+				setQueryResults( results );
+				setNoResultsFound( results?.length === 0 );
+			} );
+	}, 400 ), [] );
+
+	useEffect( () => {
+		if ( prevIDs !== attributes.restaurantIDs ) {
+			const restaurantNames = attributes.restaurantIDs.map( ( restaurantObject ) => restaurantObject.name );
+			setRidField( restaurantNames );
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( attributes.language === '' ) {
+			switch ( getLocaleCode().substring( 0, coblocksBlockData.localeCode.indexOf( '_' ) ) ) {
+				case 'fr':
+					props.setAttributes( { language: 'fr-CA' } );
+					break;
+				case 'de':
+					props.setAttributes( { language: 'de-DE' } );
+					break;
+				case 'es':
+					props.setAttributes( { language: 'es-MX' } );
+					break;
+				case 'ja':
+					props.setAttributes( { language: 'ja-JP' } );
+					break;
+				case 'nl':
+					props.setAttributes( { language: 'nl-NL' } );
+					break;
+				case 'it':
+					props.setAttributes( { language: 'it-IT' } );
+					break;
+				default:
+					props.setAttributes( { language: 'en-US' } );
+					break;
+			}
+		}
+	}, [] );
+
+	return (
+		<>
+			<Controls
+				isEditing={ isEditing }
+				setIsEditing={ setIsEditing }
+				{ ...props } />
+			<InspectorControls
+				className={ className }
+				attributes={ attributes }
+				setAttributes={ props.setAttributes }
+			/>
+			<div className={ className }>
+
+				{ ( isEditing ) ? (
+					<Placeholder
+						icon={ <Icon icon={ icon } /> }
+						label={ __( 'OpenTable', 'coblocks' ) }
+						instructions={ __(
+							'Select your OpenTable restaurant(s) to show the reservations widget.',
+							'coblocks'
+						) }
+						isColumnLayout={ true }
+					>
+
+						{ /* <form onSubmit={ renderOpenTable }> */ }
+						<div className="components-placeholder__flex-fields">
+							<FormTokenField
```

Comment:
Let's add the `maxLength` prop to this component so that a user can't search for and add multiple restaurants.

Reply:
I missed the dropdown when multiple restaurants are selected. Marking this as resolved.